### PR TITLE
feat(appsync): add Construct for AppSync HTTP DataSource

### PR DESCRIPTION
**Pull request by civilizeddev**
## Description

> https://docs.aws.amazon.com/cdk/api/latest/docs/aws-appsync-readme.html

* ADD `GraphQLApi.addHttpDataSource()`
* ADD `interface HttpDataSourceProps`
* ADD `class HttpDataSource`

## Commit Message

feat(appsync): add Construct for AppSync HTTP DataSource

## End Commit Message

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-appsync/test/appsync.test.ts [hunk 1]
```diff
@@ -2,5 +2,5 @@ import '@aws-cdk/assert/jest';
 import {} from '../lib';
 
 test('No tests are specified for this package', () => {
-  expect(true).toBe(true);
```
Comment (MrArnoldPalmer) [inline packages/@aws-cdk/aws-appsync/test/appsync.test.ts]: any reason to change this?

Comment (civilizeddev) [reply packages/@aws-cdk/aws-appsync/test/appsync.test.ts]: Just to pass `PR Linter / mandatory-changes (pull_request)` check.

I have no idea why there is only `expect(true).toBe(true);` here.

Comment (MrArnoldPalmer) [reply packages/@aws-cdk/aws-appsync/test/appsync.test.ts]: Adding an integ-test will make this rule pass as well. Just leave the unit test as is, its just a placeholder, but no need to change it if we aren't gonna actually write something useful.

**Comment by civilizeddev** [comment]
I finally learned how to write tests (integ + assert) for cdk. Please review.

**Pull request closed without merging**